c_glib/gandiva-glib/native-function.cpp

```diff
@@ -0,0 +1,209 @@
+                           ggandiva_native_function,
+                           G_TYPE_OBJECT)
+
+#define GGANDIVA_NATIVE_FUNCTION_GET_PRIVATE(obj)      \
+    static_cast<GGandivaNativeFunctionPrivate *>(      \
+        ggandiva_native_function_get_instance_private( \
+          GGANDIVA_NATIVE_FUNCTION(obj)))
+
+static void
+ggandiva_native_function_dispose(GObject *object)
+{
+  G_OBJECT_CLASS(ggandiva_native_function_parent_class)->dispose(object);
+}
+
+static void
+ggandiva_native_function_finalize(GObject *object)
+{
+  G_OBJECT_CLASS(ggandiva_native_function_parent_class)->finalize(object);
+}
+
+static void
+ggandiva_native_function_set_property(GObject *object,
+                                      guint prop_id,
+                                      const GValue *value,
+                                      GParamSpec *pspec)
+{
+  auto priv = GGANDIVA_NATIVE_FUNCTION_GET_PRIVATE(object);
+  switch (prop_id) {
+  case PROP_NATIVE_FUNCTION:
+    priv->native_function
+      = static_cast<const gandiva::NativeFunction *>(g_value_get_pointer(value));
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+ggandiva_native_function_init(GGandivaNativeFunction *object)
+{
+}
+
+static void
+ggandiva_native_function_class_init(GGandivaNativeFunctionClass *klass)
+{
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->dispose = ggandiva_native_function_dispose;
+  gobject_class->finalize = ggandiva_native_function_finalize;
+  gobject_class->set_property = ggandiva_native_function_set_property;
+
+  GParamSpec *spec;
+  spec = g_param_spec_pointer("native_function",
+                              "NativeFunction",
+                              "The raw gandiva::NativeFunction *",
+                              static_cast<GParamFlags>(G_PARAM_WRITABLE |
+                                                       G_PARAM_CONSTRUCT_ONLY));
+  g_object_class_install_property(gobject_class, PROP_NATIVE_FUNCTION, spec);
+}
+
+/**
+ * ggandiva_native_function_get_function_signature:
+ * @native_function: A #GGandivaNativeFunction.
+ *
+ * Returns: (transfer full): A #GGandivaFunctionSignature that represents
+ *   the signature of the native function.
+ */
+GGandivaFunctionSignature *
+ggandiva_native_function_get_function_signature(GGandivaNativeFunction *native_function)
+{
+  auto priv = GGANDIVA_NATIVE_FUNCTION_GET_PRIVATE(native_function);
+  auto gandiva_function_signature = priv->native_function->signature();
+  return ggandiva_function_signature_new_raw(&gandiva_function_signature);
+}
+
+/**
+ * ggandiva_native_function_get_string_function_signature:
+ * @native_function: A #GGandivaNativeFunction.
+ *
+ * Returns: (transfer full):
+ *   The string representation of the signature of the native function.
+ */
+gchar *
+ggandiva_native_function_get_string_function_signature(GGandivaNativeFunction *native_function)
```

Comment:
@kou I added this function as a useful shortcut for GLib users.
But if you don't want to add such a shortcut, I will remove it.

Reply:
If we provide this feature, we should use `ggandiva_native_function_to_string()`.